zanata-war/src/main/java/org/zanata/util/MessageGenerator.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+
+package org.zanata.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.zanata.model.HDocument;
+import org.zanata.model.HTextFlowTarget;
+
+/**
+ * Generate messages/comments of Zanata business actions
+ *
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+public class MessageGenerator {
+
+    /**
+     * Prefix of action type for generated message
+     */
+    public static final String PREFIX_MERGE_TRANS = "Merge translations";
+    public static final String PREFIX_COPY_TRANS = "Copy translation";
+    public static final String PREFIX_COPY_VERSION = "Copy version";
+
+    /**
+     * Create Zanata generated comment for merge translation
+     * @see org.zanata.service.MergeTranslationsService
+     *
+     * @param tft - HTextFlowTarget to copy from
+     */
+    public static final String getMergeTranslationMessage(
+            HTextFlowTarget tft) {
+        HDocument document = tft.getTextFlow().getDocument();
+        String author = "";
+        if (tft.getLastModifiedBy() != null) {
+            author = tft.getLastModifiedBy().getName();
+        }
+
+        return generateAutoCopiedMessage(PREFIX_MERGE_TRANS,
+                document.getProjectIteration().getProject().getName(),
+                document.getProjectIteration()
+                        .getSlug(), document.getDocId(), author);
+    }
+
+    /**
+     * Create Zanata generated comment for copy trans
+     * @see org.zanata.service.CopyTransService
+     *
+     * @param tft - HTextFlowTarget to copy from
+     */
+    public static final String getCopyTransMessage(HTextFlowTarget tft) {
+        HDocument document = tft.getTextFlow().getDocument();
+        String author = "";
+        if (tft.getLastModifiedBy() != null) {
+            author = tft.getLastModifiedBy().getName();
+        }
+        return generateAutoCopiedMessage(PREFIX_COPY_TRANS,
+                document.getProjectIteration()
+                        .getProject().getName(), document.getProjectIteration()
+                        .getSlug(), document.getDocId(), author);
+    }
+
+    /**
+     * Create Zanata generated comment for copy version
```

Comment:
In all these, is "Zanata generated" meaning that these messages are being generated by Zanata, or does it refer to the fact that the messages are saying that Zanata has generated the translations somehow? I think it might be clearer just to leave those two words out.